services/app-web/src/gqlHelpers/fetchHealthPlanPackageWrapper.ts

```diff
@@ -144,6 +144,7 @@ function useFetchHealthPlanPackageWithQuestionsWrapper(
                 },
             },
             onCompleted,
+            fetchPolicy: 'cache-and-network',
```

Comment:
> it’s not refetching clicking back and forth between questions and the summary

I believe this is because Q&A and submission summary pages live under the `SubmissionSideNav` layout route. Switching between the two pages does not unmount the `SubmissionSideNav` component and since we are doing the fetching here, it won't refetch. The routes for adding a question and response look like they are nested under the `SubmissionSideNav` layout route, but they are not, so going to these does unmount the layout route.

This question has me thinking about some work to be done with these routes. @haworku this refetching may break when we move the add question and response routes to be nested under this layout route, but I think we can then add the [onQueryUpdated](https://www.apollographql.com/docs/react/data/mutations/#refetching-after-update) option in the `createQuestion` and `createResponse` mutations to kind of do the same thing.

Reply:
@JasonLin0991  let me know if you need anymore feedback from me or want to pass me any part of this work. I want to summarize what I think we converged on for requirements for cacheing just to double check:

1. Whenever a user uploads/edits Q&A resources, we update the overall list of of questions and question responses. The main Q&A page stays up to date without manual user refresh. 
2. When a user goes to submissions summary page via sidenav and then back the Q&A page via sidenav,  we update the overall list of of questions and question response. The main Q&A page stays up to date without manual user refresh. 
3. When a user goes to QA page via sidenav and then back the summary page via sidenav, we don't need to refetch submission summary data. The user will have to refresh to get latest data and that's fine since this is a single resource  and its not a "shared" resrouce in the same way as Q&A. 
4. Avoid loading state when returning to Q&A main page from upload pages

Let  me know if this is not our shared understanding. Also, if this list feels like too much for the scope of this PR that's fine. In that case I suggest we prioritize what we are focused on.

Reply:
> 1. Whenever a user uploads/edits Q&A resources, we update the overall list of questions and question responses. The main Q&A page stays up to date without manual user refresh.

Yes! When we upload/edit Q&A resources, we are manually updating the Apollo cache on successful mutation using the `update` option, and then `onQueryUpdated` option will refetch queries that had their cache updated.

> 2. When a user goes to submissions summary page via sidenav and then back the Q&A page via sidenav, we update the overall list of questions and question response. The main Q&A page stays up to date without manual user refresh.

No, switching between the two pages does not unmount the `SubmissionSideNav` component and since we are doing the fetching here, it won't refetch. Now that `UploadQuestion` and `UploadResponse` both live under the `SubmissionSideNav` layout route, we have to use `onQueryUpdated` option in upload/edit Q&A resources mutations to force a refresh.

> 3. When a user goes to QA page via sidenav and then back the summary page via sidenav, we don't need to refetch submission summary data. The user will have to refresh to get latest data and that's fine since this is a single resource and its not a "shared" resrouce in the same way as Q&A.

Not exactly. We don't refetch switching from Q&A page back to submission summary page via the side bar nav. 

When visiting the summary or the Q&A page, wether by direct url or from the dashboard, there is one single fetch `useFetchHealthPlanPackageWithQuestionsWrapper` made to get package and questions data on the `SubmissionSideNav` component. 

This data is passed to both Q&A and submission summary pages, since `SubmissionSideNav` component is not unmounted when switching between pages, it will not refetch.

Could you explain more about what `shared` resource means?

> 4. Avoid loading state when returning to Q&A main page from upload pages

💯

Let me know if you want to zoom about this.

Reply:
Jason and I chatted. Sounds like we are only focused on #1 and #4. I misunderstood the chat above